python/ambassador/envoy/v2/v2listener.py

```diff
@@ -649,13 +645,28 @@ def __contains__(self, port: int) -> bool:
     def items(self):
         return self.listeners.items()
 
+    def get(self, port: int, use_proxy_proto: bool) -> 'V2Listener':
+        set_upp = (not port in self)
+
+        v2listener = self[port]
+
+        if set_upp:
+            v2listener.use_proxy_proto = use_proxy_proto
+        elif v2listener.use_proxy_proto != use_proxy_proto:
+            raise Exception("listener for port %d has use_proxy_proto %s, requester wants upp %s" %
+                            (v2listener.use_proxy_proto, use_proxy_proto))
```

Comment:
This seems like it could happen based on user-config.  What does the user experience when that happens?

Reply:
It actually can't happen right now, since `use_proxy_proto` can only be configured system-wide using the `ambassador` module. The `Exception` is here for protection against us trying to change that later, but doing it at the wrong level.